simtools/simtel/simtel_histograms.py

```diff
@@ -1,26 +1,548 @@
+        Please note that this value is not supposed to match the trigger rate x estimated
+        observation time, as the simulation is optimized for computational time and the energy
+        distribution assumed is not necessarily the CTAO reference cosmic-ray spectra.
+
+        Returns
+        -------
+        int:
+            total number of simulated events.
+        """
+
+        if self._total_num_triggered_events is None:
+            _, triggered_hist = self.fill_event_histogram_dicts()
+            self._total_num_triggered_events = np.round(np.sum(triggered_hist["data"]))
+            logging.debug(f"Number of triggered showers: {self._total_num_triggered_events}")
+        return self._total_num_triggered_events
+
+    def fill_event_histogram_dicts(self):
+        """
+        Get data from the total simulated event and the triggered event histograms.
+
+        Returns
+        -------
+        dict:
+            Information about the histograms with simulated events.
+        dict:
+            Information about the histograms with triggered events.
+
+        Raises
+        ------
+        HistogramIdNotFound:
+            if histogram ids not found. Problem with the file.
+        """
+        # Save the appropriate histograms to variables
+        found_one = False
+        found_two = False
+        events_histogram = None
+        triggered_events_histogram = None
+        for hist in self.histogram:
+            if hist["id"] == 1:
+                events_histogram = hist
+                found_one = True
+            elif hist["id"] == 2:
+                triggered_events_histogram = hist
+                found_two = True
+            if found_one * found_two:
+                if "triggered_events_histogram" in locals():
+                    return events_histogram, triggered_events_histogram
+        msg = "Histograms ids not found. Please check your files."
+
+        self._logger.error(msg)
+        raise HistogramIdNotFound
+
+    @property
+    def view_cone(self):
+        """
+        View cone used in the simulation.
+
+        Returns
+        -------
+        list:
+            view cone used in the simulation [min, max]
+        """
+        if self._view_cone is None:
+            self._view_cone = self.config["viewcone"] * u.deg
+        return self._view_cone
+
+    @property
+    def solid_angle(self):
+        """
+        Solid angle corresponding to the view cone.
+
+        Returns
+        -------
+        astropy.Quantity[u.sr]:
+            Solid angle corresponding to the view cone.
+        """
+        if self._solid_angle is None:
+            self._solid_angle = cone_solid_angle(self.view_cone[1]) - cone_solid_angle(
+                self.view_cone[0]
+            )
+        return self._solid_angle
+
+    @property
+    def total_area(self):
+        """
+        Total area covered by the simulated events (original CORSIKA CSCAT), i.e., area thrown.
+
```

Comment:
Good point. I added an explanation but closer to the definition of the rht flag. It might be confusing to add it here, where there is no mention on rht.

Reply:
As mentioned above, it might be better to not name it according to a tool which is not well known (it is not even part of the sim_telarray distribution)

Reply:
There is no reason to use a different definition of radius if we have access to the upper edge directly. It is no advantage to use a different definition of the maximum radius (it is in fact wrong). The only advantage of having this option is to be able to compare the results to a previously established tool (rht). This option should be used only by people who know what rht is, only by people that doubt about the results of this simtools tool and want to compare to the results using rht. Therefore, the name should not change.

Reply:
Wondering if it is needed at all then. We definitely should make sure that the result of this application works using this comparison, but do you think we need it at all later?

Reply:
I removed the support to rht now. The results of this simtools tool will always differ from the results of the rht tool by a few percentage. If that is a problem for comparison/back compatibility/report reasons, the option to rht can be reintroduced by reverting this commit  [8109ac5](https://github.com/gammasim/simtools/pull/801/commits/8109ac5cd558b6cd6b3b69a3f0cec1f9552ffb5d)

Reply:
I returned the rht functionality but with other name now. Hope that is good enough, please let me know if otherwise